spec/vmpooler/api/v1_spec.rb

```diff
@@ -417,6 +531,120 @@ def app()
         expect_json(ok = true, http = 200)
       end
 
+      it 'returns multiple VMs even when multiple instances from the same pool are requested' do
+        post "#{prefix}/vm/pool1+pool1+pool2", ''
+
+        expected = {
+          ok: true,
+          pool1: {
+            hostname: [ 'abcdefghijklmnop', 'abcdefghijklmnop' ]
+          },
+          pool2: {
+            hostname: 'qrstuvwxyz012345'
+          }
+        }
+
+        expect(last_response.body).to eq(JSON.pretty_generate(expected))
+
+        expect_json(ok = true, http = 200)
+      end
+
+      it 'returns multiple VMs even when multiple instances from multiple pools are requested' do
+        post "#{prefix}/vm/pool1+pool1+pool2+pool2+pool2", ''
```

Comment:
It's possible to do something along those lines, but it would basically be parametrizing pairs of tests across a 6 test set, which tends to look more confusing than just having the duplication in the tests.